### ISSUE TITLE
AwsBasicCredentials bug

### ISSUE THREAD

**Issue by jpdev01** [aws/aws-sdk-java-v2#4696]
### Describe the bug

I'm trying to generate client credentials, and it returns the wrong class. 
`AwsBasicCredentials.create` returns `DefaultAwsCredentialsIdentity` instead of `AwsBasicCredentials`. 

How can I resolve this?

### Expected Behavior

`AwsBasicCredentials.create` returns `AwsBasicCredentials`

### Current Behavior

fix AwsBasicCredentials.create 

### Reproduction Steps

AwsBasicCredentials basicCredentials = AwsBasicCredentials.create(
            accessKey,
            secretKey
        )

### AWS Java SDK version used

2.21.21

### JDK version used

8

### Operating System and version

Linux

**Comment by debora-ito** [comment]
@jpdev01 please share:
- a sample code we can use to reproduce, showing how you are using AwsBasicCredentials in a service client
- the full stacktrace of the error

**Comment by jpdev01** [comment]
@debora-ito news?

**Comment by debora-ito** [comment]
@jpdev01 apologies for the silence.

I can't reproduce the error with the sample code you provided. Maybe this is caused by mixed versions of SDK modules in your classpath. All SDK modules should be in the same version.  

Can you share your dependency tree? If you're using maven the command is `mvn dependency:tree`.

**Comment by jpdev01** [comment]
@debora-ito no problem :)  

In my project, we are migrating the modules to V2, but we are doing it one by one, which means we have versions in both V1 and V2.

According to the documentation, AWS does not restrict this: https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/migration-side-by-side.html

**Comment by debora-ito** [comment]
Yes, v1 and v2 can be used side-by-side, but the rule still applies: all v1 modules should be in the same v1 version, and all v2 modules should be in the same v2 version.

**Comment by jpdev01** [comment]
Interesting, I was using only version 2.21.21 (SQS and SNS), but I will evaluate. If I come across anything, I'll let you know.

**Comment by MatthewGreigPrecor** [comment]
Hit this same issue with AwsBasicCredentials.create(accessKey, secretKey) giving the wrong type (in Groovy project, worked fine in my Java projects).  Tracked it down to the AwsBasicCredentials.builder() not returning the builder for AwsBasicCredentials (not sure why it wasn't).

Was able to work around using the following:
            AwsBasicCredentials credential = AwsBasicCredentials.ANONYMOUS_CREDENTIALS.toBuilder()
                    .accessKeyId(accessKey)
                    .secretAccessKey(secretKey)
                    .build()

instead of:
        AwsBasicCredentials credential = AwsBasicCredentials.(accessKey, secretKey)

**Comment by BrentGrammer** [comment]
Like others, I'm still having the same problem. Creating AwsBasicCredentials returns a type of DefaultCredentialsIdentity which causes creating the StaticCredentialsProvider to fail since it does not take an argument type of DefaultCredentialsIdentity.

Using KMS v2.28.5 (AWS sdk v2)

I can't use anonymous credentials or get that to work. Has anyone else understood why AwsBasicCredentials returns a type of DefaultCredentialsIdentity and what could cause that?

**Comment by jpdev01** [comment]
@MatthewGreigPrecor @BrentGrammer this worked for me:
```
implementation group: 'software.amazon.awssdk', name: 'sqs', version: '2.20.85'
```
```
private StaticCredentialsProvider getCredentials() {
        AwsBasicCredentials basicCredentials = AwsBasicCredentials.create(
                x,
                x
        )

        return StaticCredentialsProvider.create(basicCredentials)
    }
```

**Comment by BrentGrammer** [comment]
@jpdev01 Thanks, but that doesn't work in my case. I am using Lucee (CFML) and AwsBasicCredentials is returning a result that is of type DefaultCredentialsIdentity even though I am passing in my access key id and secret. For some reason, something is happening where it is running some default fallback behavior, perhaps or falling into some other case for some reason and returning the wrong type, which is causing my call to StaticCredentialsProvider.create to fail.

**Comment by jpdev01** [comment]
If I recall correctly, a dependency in my project was pulling a different version of the SDK, which caused a conflict as the project got confused between the two versions of the same library.

Check if your project isn't inadvertently importing version 1 of the SDK for other services, such as SNS?

**Comment by shane-tw** [comment]
@BrentGrammer For what it's worth, I ran into the same issue and was able to get Lucee working using the below:
```cfml
<!--- We need to use getMethod/reflection. Lucee would otherwise return DefaultAwsCredentialsIdentity instead of AwsBasicCredentials, no idea why --->
<cfset LOCAL.credentials = createObject( "java", "software.amazon.awssdk.auth.credentials.AwsBasicCredentials" ).getClass().
	getMethod("create", [createObject("java", "java.lang.String").getClass(), createObject("java", "java.lang.String").getClass()]).
	invoke( javaCast("null", ""), [LOCAL.inboundMailAccessKeyId, LOCAL.inboundMailSecretKey] )>
```
whereas the below would return DefaultAwsCredentialsIdentity:
```cfml
<cfset LOCAL.credentials = createObject( "java", "software.amazon.awssdk.auth.credentials.AwsBasicCredentials" ).create( LOCAL.inboundMailAccessKeyId, LOCAL.inboundMailSecretKey )>
```

I was using OSGI modules for importing the SDK, so I *think* other stuff shouldn't be conflicting with it.
It's probably a Lucee issue/quirk rather than something this repo should fix.

**Comment by BrentGrammer** [comment]
@shane-tw Thanks, I tried casting the object which didn't make a difference, but not a more verbose procedural approach like you're showing, so I'll give that a try and update if it works.

**Comment by ali-awan** [comment]
> [@BrentGrammer](https://github.com/BrentGrammer) For what it's worth, I ran into the same issue and was able to get Lucee working using the below:
> 
> 
> <cfset LOCAL.credentials = createObject( "java", "software.amazon.awssdk.auth.credentials.AwsBasicCredentials" ).getClass().
> 	getMethod("create", [createObject("java", "java.lang.String").getClass(), createObject("java", "java.lang.String").getClass()]).
> 	invoke( javaCast("null", ""), [LOCAL.inboundMailAccessKeyId, LOCAL.inboundMailSecretKey] )>
> whereas the below would return DefaultAwsCredentialsIdentity:
> 
@shane-tw Thanks for posting this.  I am running into the exact same issue that @BrentGrammer reported.  We recently upgrade from Lucee 5.something.something to Lucee 6.[current-release], and this was the first thing to start failing.
I checked Lucee support and their dev.lucee message board, and it was reported there, with an acknowledgement, and then a useless link to trycf, and a gist, which don't address the issue and don't work.  If you go to tryCF it because dumps a system error.
So folks have reported it to Lucee with no valid response or solution, which is extremely troubling.

Anyway, I implemented your solution as-is, and our code is running again.

**Comment by ali-awan** [comment]
@shane-tw looks like the Lucee team is actively working on a fix for this, so hopefully it makes the next release.
https://luceeserver.atlassian.net/browse/LDEV-5457
